plugins/catalog/src/components/useOwnUser.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { UserEntity } from '@backstage/catalog-model';
+import { useAsync } from 'react-use';
+import { identityApiRef, useApi } from '@backstage/core';
+import { catalogApiRef } from '../plugin';
+
+/**
+ * Get the group memberships of the logged-in user.
+ */
+export function useOwnUser(): {
+  value?: UserEntity;
+  loading: boolean;
+  error?: Error;
+} {
+  const catalogApi = useApi(catalogApiRef);
+  const identityApi = useApi(identityApiRef);
+
+  // TODO: get the full entity (or at least the full entity name) from the identityApi
+  return useAsync(async () => {
+    return (await catalogApi.getEntityByName({
```

Comment:
Nit and matter of taste: can be condensed to `useAsync(() => catalogApi.get...)`